### DISCUSSION TITLE
Cannot delete directory

### DISCUSSION THREAD

**Discussion by jedi-nz** [abraunegg/onedrive#1623] [Q&A] [answered]
I am running Ubuntu 20.04

I have installed onedrive following the instructions [here](https://github.com/abraunegg/onedrive/blob/master/docs/ubuntu-package-install.md).

I have configured the OneDrive service to run as a non-root user via systemd following the instructions [here](https://github.com/abraunegg/onedrive/blob/master/docs/USAGE.md#onedrive-service-running-as-a-non-root-user-via-systemd-with-notifications-enabled-arch-ubuntu-debian-opensuse-fedora).

The sync was successful.

I have noticed that when I delete a folder from my Ubuntu onedrive directory a .Trash-1000 directory gets created and the deleted directory is moved there. Then a sync initiates which creates a .Trash-1000 directory in my onedrive online directory and the deleted directory gets synced to that.

```
Sep 08 21:33:57 orac onedrive[11509]: Uploading differences of ./.Trash-1000
Sep 08 21:33:57 orac onedrive[11509]: Uploading new items of ./.Trash-1000
```

Is this expected behaviour? I am really wanting a deleted directory to be deleted.

Thanks.

**Comment by abraunegg** [comment]
@jedi-nz 
> I have noticed that when I delete a folder from my Ubuntu onedrive directory a .Trash-1000 directory gets created and the deleted directory is moved there. Then a sync initiates which creates a .Trash-1000 directory in my onedrive online directory and the deleted directory gets synced to that

This is odd.

What this sounds like is that you have incorrectly configured your 'sync_dir' to capture your entire home directory, rather than just being your ~/OneDrive folder.

What is your current configuration (`onedrive --display-config`) when you see this issue arise?

**Reply by jedi-nz** [reply]
Thank you for your quick reply.

A bit more background that may be relevant.

I installed Ubuntu root on zfs using the installer. I decided I wanted my OneDrive directory to be its own dataset so I did 

```
zfs create rpool/USERDATA/mike_fbnju2/OneDrive
```

And checked it had the correct mountpoint

```
mike@orac:~$ zfs get mountpoint rpool/USERDATA/mike_fbnju2/OneDrive
NAME                                 PROPERTY    VALUE                SOURCE
rpool/USERDATA/mike_fbnju2/OneDrive  mountpoint  /home/mike/OneDrive  inherited from rpool/USERDATA/mike_fbnju2
```

```
mike@orac:~$ zfs get mounted rpool/USERDATA/mike_fbnju2/OneDrive
NAME                                 PROPERTY  VALUE    SOURCE
rpool/USERDATA/mike_fbnju2/OneDrive  mounted   yes      -
```

I have installed the config file and ensured the my "sync_dir" is configured correctly.

```
Configuration file successfully loaded
onedrive version                       = v2.4.13-1~np1
Config path                            = /home/mike/.config/onedrive
Config file found in config path       = true
Config option 'check_nosync'           = false
Config option 'sync_dir'               = /home/mike/OneDrive
Config option 'skip_dir'               = 
Config option 'skip_file'              = ~*|.~*|*.tmp
Config option 'skip_dotfiles'          = false
Config option 'skip_symlinks'          = false
Config option 'monitor_interval'       = 300
Config option 'min_notify_changes'     = 5
Config option 'log_dir'                = /var/log/onedrive/
Config option 'classify_as_big_delete' = 1000
Config option 'upload_only'            = false
Config option 'no_remote_delete'       = false
Config option 'remove_source_files'    = false
Config option 'sync_root_files'        = false
Selective sync 'sync_list' configured  = false
Business Shared Folders configured     = false
```
```
mike@orac:~$ systemctl --user status onedrive
● onedrive.service - OneDrive Free Client
     Loaded: loaded (/usr/lib/systemd/user/onedrive.service; enabled; vendor preset: enabled)
     Active: active (running) since Wed 2021-09-08 22:45:32 NZST; 20min ago
       Docs: https://github.com/abraunegg/onedrive
   Main PID: 150994 (onedrive)
     CGroup: /user.slice/user-1000.slice/user@1000.service/onedrive.service
             └─150994 /usr/bin/onedrive --monitor

Sep 08 22:51:48 orac onedrive[150994]: Deleting item from OneDrive: .Trash-1000/files
Sep 08 22:51:48 orac onedrive[150994]: Deleting item from OneDrive: ./.Trash-1000
Sep 08 22:54:04 orac onedrive[150994]: Uploading differences of ./.Trash-1000
Sep 08 22:54:04 orac onedrive[150994]: Uploading new items of ./.Trash-1000
```

Your assistance in getting this resolved would be much appreciated.

**Comment by abraunegg** [comment, marked as answer]
@jedi-nz 
Glad to see someone using ZFS (heavy ZFS advocate here and OpenZFS contributor as well) .. Ubuntu .. IMHO not a wise choice.

Essentially this is your OS causing this - due to how Gnome Trash has been implemented. 

Your options to resolve are:
1. Live with it (sorry)
2. When deleting objects, use 'shift+delete' rather than the 'delete' key only - as this is what causes the problem to use the Trash folder.
3. Configure the client to skip this directory from being synced - see https://github.com/abraunegg/onedrive/blob/master/docs/USAGE.md#skip_dir
4. Globally disable this feature following this process: [https://github.com/Cordic77/libgio-disable-trash/](http://htmlpreview.github.io/?https://github.com/Cordic77/libgio-disable-trash/blob/master/Globally%20disable%20GNOME%27s%20Trash%20in%20Debian-based%20distributions.htm)
5. Submit a bug to Debian | Ubuntu to get them to 'fix' this ...

**Reply by jedi-nz** [reply]
Thank you for taking the time to reply. I went with the skip directory option.

Thank you also to take the time to create this software, it is working perfectly for me.

Do you mind if I ask what distribution you would recommend for zfs on root? Ubuntu seems like it still has some way to go in this area from my experience. I would be interested in your opinion of the best option for zfs on root.

**Reply by abraunegg** [reply]
> Do you mind if I ask what distribution you would recommend for zfs on root? Ubuntu seems like it still has some way to go in this area from my experience. I would be interested in your opinion of the best option for zfs on root.

I do not use ZFS for root - I figure that I can reinstall my main OS if needed, it is the data I care about. I would look at Arch, Debian or Gentoo - but read the Arch document (https://wiki.archlinux.org/title/Install_Arch_Linux_on_ZFS) and amend as needed for your chosen distro.

**Reply by jedi-nz** [reply]
Do you know how to solve this?

```°W: An error occurred during the signature verification. The repository is not updated and the previous index files will be used. GPG error: https://download.opensuse.org/repositories/home:/npreining:/debian-ubuntu-onedrive/xUbuntu_21.04 ./ InRelease: The following signatures were invalid: EXPKEYSIG B8AC39B0876D807E home:npreining OBS Project <home:npreining@build.opensuse.org>
W: Failed to fetch https://download.opensuse.org/repositories/home:/npreining:/debian-ubuntu-onedrive/xUbuntu_21.04/./InRelease  The following signatures were invalid: EXPKEYSIG B8AC39B0876D807E home:npreining OBS Project <home:npreining@build.opensuse.org>
W: Some index files failed to download. They have been ignored, or old ones used instead.
```

**Reply by abraunegg** [reply]
> Do you know how to solve this?
> 
> ```
> W: Failed to fetch https://download.opensuse.org/repositories/home:/npreining:/debian-ubuntu-onedrive/xUbuntu_21.04/./InRelease  The following signatures were invalid: EXPKEYSIG B8AC39B0876D807E home:npreining OBS Project <home:npreining@build.opensuse.org>
> W: Some index files failed to download. They have been ignored, or old ones used instead.
> ```

Your platform is having issues with the security keys - please review https://github.com/abraunegg/onedrive/discussions/2000 for ideas .. this is not an issue with the client but your OS and how certificates are being managed

**Comment by jedi-nz** [comment]
Hi,

After upgrading to Ubuntu 24.04 I am trying to re-install onedrive.

Can you tell me how to solve this?

```
mike@orac:~$ sudo apt install --no-install-recommends --no-install-suggests onedrive
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
Some packages could not be installed. This may mean that you have
requested an impossible situation or if you are using the unstable
distribution that some required packages have not yet been created
or been moved out of Incoming.
The following information may help to resolve the situation:

The following packages have unmet dependencies:
 onedrive : Depends: libphobos2-ldc-shared100 (>= 1:1.30.0) but it is not installable
E: Unable to correct problems, you have held broken packages.
mike@orac:~$ sudo dpkg --configure -a
```
Thank you.

**Comment by abraunegg** [comment]
@jedi-nz 
Please re-read the iinstructions on how to install this client. 

You are not using the Ubuntu 24.x instructions

**Comment by jedi-nz** [comment]
I was following the directions here?

https://github.com/abraunegg/onedrive/blob/master/docs/ubuntu-package-install.md#distribution-ubuntu-2404

**Comment by jedi-nz** [comment]
Sorry my mistake now its working, not sure what I did wrong.

**Reply by abraunegg** [reply]
100% you used the wrong instructions to begin with, leading to your issue.
